Add support for unassociated streams

idastambuk opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. To surface this PR in the changelog add the label: changelog
    If this PR is going in the changelog please make sure the title of the PR explains the feature in a user-centric way:
        Bad: fix state bug in hooks
        Good: Fix crash when switching from Query Builder

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, consider updating the documentation in README.md(https://github.com/grafana/iot-sitewise-datasource/blob/main/README.md) or [src/README.md](https://github.com/grafana/iot-sitewise-datasource/blob/main/README.md).

-->

**What this PR does / why we need it**:
When querying by propertyAlias was first implemented, it had support for unassociated streams ("disassociated" in AWS). 
Unassociated streams are data streams that are not associated with a propertyId on an asset, but contain historical data about a property all the same. They can be queried by a propertyAlias. 
You can see them here in the console if you apply the Disassociated Streams filter: 

<img width="800" alt="Screenshot 2023-10-13 at 7 13 43 PM" src="https://github.com/grafana/iot-sitewise-datasource/assets/16140639/bcfb6115-7b8a-4552-9e71-ffeb965d46f4">

[The 500 error from the bug report](https://github.com/grafana/iot-sitewise-datasource/issues/228) happened because of a null pointer reference since, when we query DescribeAssetPropertyWithContext for unassociated propertyAliases, it doesn't return an assetId or propertyId. The bug was introduced when we implemented fetching assets and property values only by propertyAlias. However, this only worked for associated streams, where DescribeAssetPropertyWithContext will actually return propertyId.

```
func getAssetIdAndPropertyId(query models.AssetPropertyValueQuery, client client.SitewiseClient, ctx context.Context) (models.AssetPropertyValueQuery, error) {
	result := query
	if query.PropertyAlias != "" {
		resp, err := client.DescribeTimeSeriesWithContext(ctx, &iotsitewise.DescribeTimeSeriesInput{
			Alias: getPropertyAlias(query.BaseQuery),
		})
		if err != nil {
			return models.AssetPropertyValueQuery{}, err
		}
		---> result.AssetIds = []string{*resp.AssetId}
		result.PropertyId = *resp.PropertyId
		
```
<img width="1422" alt="Screenshot 2023-10-16 at 12 13 14 PM" src="https://github.com/grafana/iot-sitewise-datasource/assets/16140639/c9f6fbff-0f9a-442a-af38-c03d8502267e">


On the frontend, I fixed the panel header to show propertyAlias in the panel header when collapsed
<img width="948" alt="Screenshot 2023-10-16 at 12 11 40 PM" src="https://github.com/grafana/iot-sitewise-datasource/assets/16140639/d7996deb-95ba-48e1-be12-d0054650697c">

I also moved some functions in the util package from /api, since I needed them in other packages

**Which issue(s) this PR fixes**:


### How to test: 
Add the following propertyAliased into the property alias field and test different "Get value ..." query types. For data, select "last year" in order to be able to see data, then you can zoom in. 

We have a few aliased streams in our dev account: 
turbine/1/torque -> disassociated - doesn't have an assetId/propertyId tied to it, so this can be tested for a fix
/turbine/1/wind_direction -> associated - for comparison
<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #

**Special notes for your reviewer**:

fridgepoet commented:
Heya @idastambuk just trying to understand what is expected here. I've tried "Get property value history" on some dissociated data streams in our squad dev account, but I'm getting a panic in the backend code still. This seems to be the same as what's currently on v1.11.1 
I think it's panicking here https://github.com/grafana/iot-sitewise-datasource/pull/231/files#diff-670b06752c76d1a93e8f926fe7568b7de2bce83536b701c5de5d723a94e4676eR27
Is that expected?
Let me know if I've got something wrong or missed something!

ex:
![Screenshot 2023-10-17 at 18 08 53](https://github.com/grafana/iot-sitewise-datasource/assets/4163034/baf9619f-03fe-45b0-bf24-6ec309f8c585)

idastambuk commented:
> Heya @idastambuk just trying to understand what is expected here. I've tried "Get property value history" on some dissociated data streams in our squad dev account, but I'm getting a panic in the backend code still. This seems to be the same as what's currently on v1.11.1 I think it's panicking here as `h` is empty https://github.com/grafana/iot-sitewise-datasource/pull/231/files#diff-670b06752c76d1a93e8f926fe7568b7de2bce83536b701c5de5d723a94e4676eR53 Is that expected? Let me know if I've got something wrong or missed something!
> 
> ex: ![Screenshot 2023-10-17 at 18 08 53](https://user-images.githubusercontent.com/4163034/275903637-baf9619f-03fe-45b0-bf24-6ec309f8c585.png)

Hey @fridgepoet, good catch, it only happens with property value history queries and when there is no data. I added a condition and updated the tests.

idastambuk commented on pkg/framer/property_value.go:
```diff
@@ -19,9 +19,9 @@ func (p AssetPropertyValue) Frames(ctx context.Context, resources resource.Resou
 		return nil, err
 	}
-
+	var property *iotsitewise.DescribeAssetPropertyOutput
```
Who knows what it did, but fixed it now!🙏🏻

idastambuk commented on pkg/server/test/property_value_aggregate_test.go:
```diff
@@ -195,4 +196,113 @@ func TestPropertyValueAggregate(t *testing.T) {
+		expectedDescribeTimeSeriesWithContextArgs: &iotsitewise.DescribeTimeSeriesInput{Alias: Pointer("/amazon/renton/1/rpm")},
+		expectedMaxPages:   1,
+		expectedMaxResults: 0,
+		isDisassociated:    true,
```
Good catch!
